StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheet+API.swift

```diff
@@ -582,44 +591,103 @@ extension PaymentSheet {
         return params
     }
 
-    private static func _handleIntentCreation(_ result: Result<String, Error>) {
-        // TODO(porter) Deferred intent confirmation handling
+    private static func createPaymentMethodIfNeeded(apiClient: STPAPIClient,
+                                                    paymentMethod: STPPaymentMethod?,
+                                                    paymentMethodParams: STPPaymentMethodParams?) async throws -> STPPaymentMethod {
+        if let paymentMethod = paymentMethod {
+            return paymentMethod
+        }
+
+        guard let paymentMethodParams = paymentMethodParams else {
+            throw PaymentSheetError.unknown(debugDescription: "paymentMethodParams unexpectedly nil")
+        }
+
+        return try await apiClient.createPaymentMethod(with: paymentMethodParams)
     }
 
-    private static func confirmDeferredIntentNewPaymentMethod(configuration: PaymentSheet.Configuration,
-                                                              intentConfig: IntentConfiguration,
-                                                              paymentMethodParams: STPPaymentMethodParams,
-                                                              shouldSavePaymentMethod: Bool,
-                                                              completion: @escaping (PaymentSheetResult) -> Void) {
-        configuration.apiClient.createPaymentMethod(with: paymentMethodParams) {
-            paymentMethod,
-            error in
-            if let error = error {
-                completion(.failed(error: error))
-                return
-            }
+    private static func handleDeferredIntentConfirmation(deferredIntentContext: DeferredIntentContext,
+                                                         paymentMethod: STPPaymentMethod?,
+                                                         paymentMethodParams: STPPaymentMethodParams?,
+                                                         shouldSavePaymentMethod: Bool) {
+        Task {
+            do {
+                // Create PM if necessary
+                let _paymentMethod = try await createPaymentMethodIfNeeded(apiClient: deferredIntentContext.configuration.apiClient,
```

Comment:
Why _paymentMethod?

Ah to avoid overwriting `paymentMethod`?  Can you do `let paymentMethod = paymentMethod ?? try await ...`

Reply:
I'd really like to do this but putting the complier doesn't like it, that's why I opted to create a helper method. There are some other ways to write it but I prefer this, the suggestions in the SO post are confusing to read IMO.

https://stackoverflow.com/questions/68625044/swift-5-5-async-operation-as-optional-default
![CleanShot 2023-03-09 at 22 45 08](https://user-images.githubusercontent.com/88012362/224233730-45a2c749-b05c-4005-833f-c8ceb8684783.png)

Reply:
Ah okay.  Let's just remove the `_`?

Reply:
removed